external/storm-hive/src/main/java/org/apache/storm/hive/bolt/HiveBolt.java

```diff
@@ -56,9 +56,9 @@
     private ExecutorService callTimeoutPool;
     private transient Timer heartBeatTimer;
     private Boolean kerberosEnabled = false;
-    private AtomicBoolean timeToSendHeartBeat = new AtomicBoolean(false);
+    private Boolean sendHeartBeat = true;
```

Comment:
I think this needs to be volatile or atomic.  It is written in one thread and read in another with no locks around any of it.